modules/mapper-extras/src/main/java/org/opensearch/index/mapper/ScaledFloatFieldMapper.java

```diff
@@ -229,7 +229,7 @@ public String typeName() {
         @Override
         public Query termQuery(Object value, QueryShardContext context) {
             failIfNotIndexedAndNoDocValues();
-            long scaledValue = Math.round(scale(value));
+            long scaledValue = Math.round(parse(value) * scalingFactor);
```

Comment:
Consistent with precision loss when indexing

Reply:
We can also changing the code in indexing, but it will cause indexing performance regression.
```
Math.round(doubleValue * scalingFactor) -> Math.round(scale(value))
```

Reply:
This look good to me, it's better to extract this logic `Math.round(parse(value) * scalingFactor)` to a single place, so termQuery and parseCreateField for indexing just use that.